sale_confirm_group/models/sale_order.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, exceptions, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    user_can_confirm = fields.Boolean(
+        compute="_compute_user_can_confirm", compute_sudo=True
+    )
+
+    @api.depends(
+        # Keep these dotted fields as dependencies, because:
+        # - chances of them being updated very often (leading to lots of records cache
+        #   invalidations and field recomputation) are quite low
+        # - removing them in favor of ``company_id`` alone (or no deps at all) means
+        #   we'd have to invalidate the records cache manually when we need to check
+        #   whether a user can actually confirm a sale order (to avoid using an outdated
+        #   value for this field, since settings may have been updated in the meanwhile)
+        "company_id.use_sale_confirmation_groups",
+        "company_id.sale_confirmation_group_ids.users",
+    )
+    @api.depends_context("uid")
+    def _compute_user_can_confirm(self):
+        user = self.env.user
+        for company, sales in self.grouped("company_id").items():
+            if not company.use_sale_confirmation_groups:
+                sales.user_can_confirm = True
+            elif not (groups := company.sale_confirmation_group_ids):
+                sales.user_can_confirm = True
+            else:
+                sales.user_can_confirm = user in groups.users
```

Comment:
We could have an archived user processing orders (e.g. OdooBot with `sale_automatic_workflow` modue), and `user in groups.users` check won't work in that case:
```suggestion
                sales.user_can_confirm = user in groups.with_context(active_test=False).users
```

Reply:
I adressed this issue in the 2nd commit.